test/bigtable/filter.js

```diff
@@ -0,0 +1,677 @@
+        fail: { c: 'c' }
+      };
+
+      Filter.parse = spy(function(value) {
+        return value;
+      });
+
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'condition');
+        assert.deepEqual(value, {
+          predicateFilter: condition.test,
+          trueFilter: condition.pass,
+          falseFilter: condition.fail
+        });
+
+        var parseCalls = Filter.parse.calls;
+        assert.strictEqual(parseCalls[0].args[0], condition.test);
+        assert.strictEqual(parseCalls[1].args[0], condition.pass);
+        assert.strictEqual(parseCalls[2].args[0], condition.fail);
+
+        Filter.parse = parse;
+        done();
+      };
+
+      filter.condition(condition);
+    });
+  });
+
+  describe('family', function() {
+    it('should create a family name regex filter', function(done) {
+      var convert = Filter.convertToRegExpString;
+      var familyName = 'fake-family';
+
+      Filter.convertToRegExpString = spy(function(value) {
+        return value;
+      });
+
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'familyNameRegexFilter');
+        assert.strictEqual(value, familyName);
+        assert.strictEqual(
+          Filter.convertToRegExpString.calls[0].args[0], familyName);
+
+        Filter.convertToRegExpString = convert;
+        done();
+      };
+
+      filter.family(familyName);
+    });
+  });
+
+  describe('interleave', function() {
+    it('should create an interleave filter', function(done) {
+      var fakeFilters = [{}, {}, {}];
+      var parse = Filter.parse;
+
+      Filter.parse = spy(function(value) {
+        return value;
+      });
+
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'interleave');
+        assert.strictEqual(value.filters[0], fakeFilters[0]);
+        assert.strictEqual(value.filters[1], fakeFilters[1]);
+        assert.strictEqual(value.filters[2], fakeFilters[2]);
+
+        var parseCalls = Filter.parse.calls;
+        assert.strictEqual(parseCalls[0].args[0], fakeFilters[0]);
+        assert.strictEqual(parseCalls[1].args[0], fakeFilters[1]);
+        assert.strictEqual(parseCalls[2].args[0], fakeFilters[2]);
+
+        Filter.parse = parse;
+        done();
+      };
+
+      filter.interleave(fakeFilters);
+    });
+  });
+
+  describe('label', function() {
+    it('should apply the label transformer', function(done) {
+      var label = 'label';
+
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'applyLabelTransformer');
+        assert.strictEqual(value, label);
+        done();
+      };
+
+      filter.label(label);
+    });
+  });
+
+  describe('row', function() {
+    it('should apply the row key regex filter', function(done) {
+      var rowKey = 'gwashington';
+      var convert = Filter.convertToRegExpString;
+
+      Filter.convertToRegExpString = spy(function(value) {
+        return value;
```

Comment:
I think we can end up with false positives by returning the same value that was passed in. When we check the resulting value, we compare it to the original, which means while we can confirm the override was called, we aren't confirming its work made any difference.